docker/install-dynamic-plugins.py

```diff
@@ -100,21 +102,88 @@ def maybeMergeConfig(config, globalConfig):
     else:
         return globalConfig
 
-def mergePlugin(plugin: dict, allPlugins: dict, dynamicPluginsFile: str):
+def mergePlugin(plugin: dict, allPlugins: dict, dynamicPluginsFile: str, level: int):
     package = plugin['package']
+    version = None # Used to track OCI package versions for version inference
     if not isinstance(package, str):
         raise InstallException(f"content of the \'plugins.package\' field must be a string in {dynamicPluginsFile}")
 
+    if package.startswith('oci://'):
+        # Check if package is in one of the expected formats: 
+        # Tag format: `oci://<registry>:<tag>!<path>`
+        # SHA digest format: `oci://<registry>@sha<algo>:<digest>!<path>`
+        oci_pattern = (
+            r'^(oci://[^\s:@]+)'
+            r'(?:'
+                r':([^\s!@:]+)'  # tag only
+                r'|'
+                r'@((?:sha256|sha512|blake3):[^\s!@:]+)'  # digest only
+            r')'
+            r'!([^\s]+)$'
+        )
+        match = re.match(oci_pattern, package)
+        if not match:
+            raise InstallException(f"oci package \'{package}\' is not in the expected format \'oci://<registry>:<tag>!<path>\' or \'oci://<registry>@sha<algo>:<digest>!<path>\' in {dynamicPluginsFile} where <algo> is one of {RECOGNIZED_ALGORITHMS}")
+
+        # Strip away the version (tag or digest) from the package string, resulting in oci://<registry>:!<path>
+        # This helps ensure keys used to identify OCI plugins are independent of the version of the plugin
+        registry = match.group(1)
+        tag_version = match.group(2)
+        digest_version = match.group(3)
+
+        version = tag_version if tag_version else digest_version
+            
+        path = match.group(4) 
+        
+        # {{inherit}} tag indicates that the version should be inherited from the included configuration. Must NOT have a SHA digest included.
+        inheritVersion = (tag_version == "{{inherit}}" and digest_version == None)
+        package = f"{registry}:!{path}"
+
     # if `package` already exists in `allPlugins`, then override its fields
     if package not in allPlugins:
+        if package.startswith('oci://'):
+            if inheritVersion is True:
```

Comment:
Is there any way to reduce the nested `if` statements here and on line 165-166?  This kind of nesting tends to lead to difficult to follow logic, and if we make this kind of test more than once maybe it'd be more clear in intent to have a helper function that makes this test.

Reply:
@gashcrumb do you consider it resolved ?